lib/teleagent/agent.go

```diff
@@ -85,9 +85,10 @@ func (a *AgentServer) Serve() error {
 				return trace.Wrap(err, "unknown error")
 			}
 			if !neterr.Temporary() {
-				if !strings.Contains(neterr.Error(), "use of closed network connection") {
-					log.Errorf("got permanent error: %v", err)
+				if strings.Contains(neterr.Error(), "use of closed network connection") {
+					return nil
 				}
+				log.Errorf("got permanent error: %v", err)
```

Comment:
Nit: We started trying to make all log entries into proper sentences some time ago which includes capitalizing log messages and ending them with a dot, so I'd refactor this one as well since you're here anyway:

```go
log.WithError(err).Error("Got permanent error.")
```

Reply:
Done.